api/src/main/java/jakarta/servlet/http/HttpServlet.java

```diff
@@ -317,10 +305,18 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws Se
     protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         String protocol = req.getProtocol();
         String msg = lStrings.getString("http.method_delete_not_supported");
-        if (protocol.endsWith("1.1")) {
-            resp.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED, msg);
-        } else {
+        switchProtocol(protocol, resp, msg);
+    }
+
+    private static void switchProtocol(String protocol, HttpServletResponse resp, String msg) throws IOException {
+        switch (protocol) {
+        case "HTTP/0.9":
+        case "HTTP/1.0":
             resp.sendError(HttpServletResponse.SC_BAD_REQUEST, msg);
+            break;
+        default:
+            resp.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED, msg);
+            break;
         }
```

Comment:
Make non-static protected so that it can be overridden
return the code rather than apply it.
```suggestion
    protected int getMethodNotSupportedCode(String protocol) {
        switch (protocol) {
          case "HTTP/0.9":
          case "HTTP/1.0":
            return HttpServletResponse.SC_BAD_REQUEST;
        default:
            return HttpServletResponse.SC_METHOD_NOT_ALLOWED;
        }
```